src/registry/AvatarRegistry.sol

```diff
@@ -0,0 +1,523 @@
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address to be cancel upkeep
+    function cancelAvatarUpKeep(address avatarAddress) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+        // NOTE: only avatar which upkeep is being cancelled can be removed
+        uint256 upKeepId = avatarsInfo[avatarAddress].upKeepId;
+        CL_REGISTRY.cancelUpkeep(upKeepId);
+    }
+
+    /// @dev Updates status of an avatar in the registry
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address update status of
+    /// @param newStatus latest status of the avatar
+    function updateStatus(address avatarAddress, AvatarStatus newStatus) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+
+        AvatarStatus oldStatus = avatarsInfo[avatarAddress].status;
+        if (oldStatus == newStatus) {
+            revert UpdateSameStatus();
+        }
+
+        avatarsInfo[avatarAddress].status = newStatus;
+
+        emit UpdateAvatarStatus(avatarAddress, oldStatus, newStatus, block.timestamp);
+    }
+
+    /// @dev Withdraws LINK funds and remove avatar from registry
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address to be remove from registry
+    function withdrawLinkFundsAndRemoveAvatar(address avatarAddress) external onlyGovernance {
+        if (!_avatars.contains(avatarAddress)) {
+            revert NotAvatarIncluded(avatarAddress);
+        }
+
+        uint256 upKeepId = avatarsInfo[avatarAddress].upKeepId;
+        // NOTE: only avatar which upkeep is being cancelled can be removed
+        (,,,,,, uint64 maxValidBlocknumber,) = CL_REGISTRY.getUpkeep(upKeepId);
+        // https://etherscan.io/address/0x02777053d6764996e594c3e88af1d58d5363a2e6#code#F1#L738
+        if (maxValidBlocknumber == type(uint64).max) {
+            revert UpKeepNotCancelled(upKeepId);
+        }
+
+        // NOTE: removal actions after on-chain checkups
+        _avatars.remove(avatarAddress);
+        delete avatarsInfo[avatarAddress];
+
+        CL_REGISTRY.withdrawFunds(upKeepId, address(this));
+
+        emit RemoveAvatar(avatarAddress, upKeepId, block.timestamp);
+    }
+
+    /// @dev  Sweep the full LINK balance to techops
+    function sweepLinkFunds() external onlyGovernance {
+        uint256 linkBal = LINK.balanceOf(address(this));
+        LINK.transfer(TECHOPS, linkBal);
+        emit SweepLinkToTechops(linkBal, block.timestamp);
+    }
+
+    /// @dev  Sweep the full ETH balance to recipient
+    /// @param recipient Address receiving eth funds
+    function sweepEthFunds(address payable recipient) external onlyGovernance {
+        uint256 ethBal = address(this).balance;
+        recipient.transfer(ethBal);
+        emit SweepEth(recipient, ethBal);
+    }
```

Comment:
why does this have a recipient and `sweepLinkFunds` does not?

Reply:
Yeah, and why does one event have a timestamp while the other doesn't? I would also just change `sweepLinkFunds` to sweep any erc20